src/App.test.jsx

```diff
@@ -1,34 +1,107 @@
 import React from 'react';
+import {
+  MemoryRouter,
+} from 'react-router-dom';
 
 import { render } from '@testing-library/react';
 
 import { useDispatch, useSelector } from 'react-redux';
 
 import App from './App';
 
-test('App', () => {
-  const dispatch = jest.fn();
+const customRender = (path) => render((
+  <MemoryRouter initialEntries={[path]}>
+    <App />
+  </MemoryRouter>
+));
 
-  useDispatch.mockImplementation(() => dispatch);
+describe('App', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
 
-  useSelector.mockImplementation((selector) => selector({
-    regions: [
-      { id: 1, name: '서울' },
-    ],
-    categories: [
-      { id: 1, name: '한식' },
-    ],
-    restaurants: [
-      { id: 1, name: '마법사주방' },
-    ],
-  }));
+    const dispatch = jest.fn();
+    useDispatch.mockImplementation(() => dispatch);
 
-  const { queryByText } = render((
-    <App />
-  ));
+    useSelector.mockImplementation((selector) => selector({
+      regions: [
+        { id: 1, name: '서울' },
+      ],
+      categories: [
+        { id: 1, name: '한식' },
+      ],
+      restaurants: [
+        { id: 1, name: '한국식 초밥' },
+      ],
+      selectedRestaurant: { id: 1, name: '한국식 초밥' },
+      restaurant: {
+        id: 6,
+        categoryId: 1,
+        name: '한국식 초밥',
+        address: '서울 강남구',
+        menuItems: [
+          {
+            id: 18,
+            restaurantId: 6,
+            name: '밥',
+          },
+        ],
+      },
+    }));
+  });
+
+  context('with path /', () => {
+    it('shows header', () => {
+      const { queryByText } = customRender('/');
+
+      expect(queryByText('헤더')).not.toBeNull();
+    });
+
+    it('shows HOME heading', () => {
```

Comment:
```suggestion
    it('shows "HOME" heading', () => {
```
개인적으로는 특정 문자열이 나오길 기대하는거라면 문자열 이라는걸 보여주는게 오해가 생기지 않는거 같아요 :)

Reply:
넵 수정하였습니다.